### DISCUSSION TITLE
Query: Few questions on usage of Karavan 4.x

### DISCUSSION THREAD

**Discussion by naqvis** [apache/camel-karavan#1004] [Q&A] [answered]
Thank you for awesome work 👍 on Camel Karavan.
 
I've deployed Karavan 4.1.1 on Kubernetes running on K3s. Few questions I am seeking guidance are:

* How can we enable 3.x like CI/CD behavior?
* How can we toggle the data (for red square area of below snapshot)?
<img width="758" alt="image" src="https://github.com/apache/camel-karavan/assets/10077630/92f8447d-3d36-49fb-969b-c3e860b6be05">

* Why Trace panel is blank when  project running in dev mode? (snapshot below)
<img width="859" alt="image" src="https://github.com/apache/camel-karavan/assets/10077630/7cda653c-eee2-45ef-90aa-9de249c7a7f8">

* How can we enable quarkus and spring run-times when creating project on GUI? (3.x used to have this option). I've tried with setting env var `KARAVAN_RUNTIMES` with list of run times, but that doesn't seem to work on 4.x
* What extra services and/or setup is required for `Build` and `RollOut` buttons to function properly?
* Using same Github repo (I've configured the Karavan secret with details of external Github Repo for projects storage), yields different view of projects (some times it display few projects, some times it just doesn't display any of the project). I've looked at the container and github repo was properly cloned and contained all projects and their artefacts
* Previous versions like 3.x were bundled with `Tekton` to deal with build + push + deployment of project(s), but I'm not seeing that getting installed by the installer. Is CI/CD part of 3.x deprecated? or users are required to setup Tekton on their own?

Thanks in advance

**Comment by mgubaidullin** [comment, marked as answer]
1. Karavan 4.x supports camel-main runtime. We do not have manpower for now to support all runtimes

2. Karavan 4.x uses camel-karavan-devmode to build and deploy integrations in Docker or Kubernetes.
You can customize *-build.sh script in your `templates` project to call any CI/CD

**Comment by mgubaidullin** [comment]
The latest Karavan release is 4.1.0. 
4.1.1 is work-in-progress.
https://github.com/apache/camel-karavan/releases

**Comment by naqvis** [comment]
Thank you @mgubaidullin and appreciate your response.
